cognite/extractorutils/_extensions.py

```diff
@@ -0,0 +1,187 @@
+
+
+class BaseExtensionExtractor(Extractor[CustomConfigClass]):
+    """
+    Base class for extension extractors.
+
+    Intended for use with an extractor that is capable of producing CdfTypes from some underlying source.
+
+    Args:
+        name: Name of the extractor, how it's invoked from the command line.
+        description: A short 1-2 sentence description of the extractor.
+        version: Version number, following semantic versioning.
+        run_handle: A function to call when setup is done that runs the extractor, taking a cognite client, state store
+            config object and a shutdown event as arguments.
+        config_class: A class (based on the BaseConfig class) that defines the configuration schema for the extractor
+        metrics: Metrics collection, a default one with be created if omitted.
+        use_default_state_store: Create a simple instance of the LocalStateStore to provide to the run handle. If false
+            a NoStateStore will be created in its place.
+        cancelation_token: An event that will be set when the extractor should shut down, an empty one will be created
+            if omitted.
+        config_file_path: If supplied, the extractor will not use command line arguments to get a config file, but
+            rather use the supplied path.
+        continuous_extractor: If True, extractor will both successful start and end time. Else, only show run on exit.
+        heartbeat_waiting_time: Time interval between each heartbeat to the extraction pipeline in seconds.
+    """
+
+    def __init__(
+        self,
+        *,
+        name: str,
+        description: str,
+        version: Optional[str] = None,
+        run_handle: Optional[Callable[[CogniteClient, AbstractStateStore, CustomConfigClass, Event], None]] = None,
+        config_class: Type[CustomConfigClass],
+        metrics: Optional[BaseMetrics] = None,
+        use_default_state_store: bool = True,
+        cancelation_token: Event = Event(),
+        config_file_path: Optional[str] = None,
+        continuous_extractor: bool = False,
+        heartbeat_waiting_time: int = 600,
+        handle_interrupts: bool = True,
+    ):
+        super(BaseExtensionExtractor, self).__init__(
+            name=name,
+            description=description,
+            version=version,
+            run_handle=run_handle,
+            config_class=config_class,
+            metrics=metrics,
+            use_default_state_store=use_default_state_store,
+            cancelation_token=cancelation_token,
+            config_file_path=config_file_path,
+            continuous_extractor=continuous_extractor,
+            heartbeat_waiting_time=heartbeat_waiting_time,
+            handle_interrupts=handle_interrupts,
+        )
+
+        self._event_queue_size = 10_000
+        self._raw_queue_size = 100_000
+        self._timeseries_queue_size = 1_000_000
+        self._upload_interval = 60
```

Comment:
These should either be configurable (by having a `ExtensionConfig` class that we use as base for the generic instead of `BaseConfig` with these in an `extractor` section or similar), or as module level constants IMO

Reply:
Good suggestion, a slightly extended config class somehow seems reasonable.